### PR TITLE
Doc alternative space array package

### PR DESCRIPTION
**Pull request by PallHaraldsson**
[skip ci]

### CODE REVIEW

**Comment by PallHaraldsson** [comment]
In the docs, I couldn't press the edit, to well edit, and the search in the codebase wasn't helpful, then I remembered moved to stdlib. Maybe something can be done about that, for the link and/or doc better.

I found Jutho's package in discourse thread. I'm assuming it works well, at least it's good to know of, and limitations in stdlib should be doced anyway. I don't know if it's the best one (or e.g. as good as Python's sparse), I found other packages thay may not actually be related: https://juliahub.com/ui/Packages/LuxurySparse/4Qz8z/0.6.1

**Comment by Wimmerer** [comment]
Fantastic! While I'm not sure what is unique about the LuxurySparse package the other two (ExtendableSparse and SparseArrayKit) should at least have a little description. After that we're good to merge!

**Comment by PallHaraldsson** [comment]
LuxurySparse has some additional types so I added it too, and one more that seemed helpful.

I would argue for merging the original commit too, as it clarifies limitation, or at least that part if you do not want to link to a package there. You could however add to that, "see package below" (maybe with a link to that section).

File: docs/src/index.md [hunk 1]
```diff
@@ -5,9 +5,12 @@ DocTestSetup = :(using SparseArrays, LinearAlgebra)
...
 Julia has support for sparse vectors and [sparse matrices](https://en.wikipedia.org/wiki/Sparse_matrix)
-in the `SparseArrays` stdlib module. Sparse arrays are arrays that contain enough zeros
-that storing them in a special data structure leads to savings in space and execution time,
-compared to dense arrays.
+in the `SparseArrays` stdlib module, but limted to (1D and) 2D arrays, unlike the regular dense arrays.
+There's an external package [SparseArrayKit.jl](https://github.com/Jutho/SparseArrayKit.jl) available
+for multidimensional sparse arrays (and also other packages for sparse).
+
+Sparse arrays are arrays that contain enough zeros that storing them in a special data structure leads
+to savings in space and execution time, compared to dense arrays.
```
Comment (Wimmerer) [inline docs/src/index.md:13]: The first line (sparse vectors and sparse matrices) implies 1D and 2D. 
```suggestion
in the `SparseArrays` stdlib module. Sparse arrays are arrays that contain enough zeros that storing them in a special data structure leads to savings in space and execution time, compared to dense arrays.

External packages which implement different sparse storage types, multidimensional sparse arrays, and more can be found in [Noteworthy external packages](@ref man-csc)
```

File: docs/src/index.md [hunk 2]
```diff
@@ -249,3 +252,9 @@ Several other Julia packages provide sparse matrix implementations that should b
 3. [SparseMatricesCSR.jl](https://github.com/gridap/SparseMatricesCSR.jl) provides a Julia native implementation of the Compressed Sparse Rows (CSR) format.
 
 4. [MKLSparse.jl](https://github.com/JuliaSparse/MKLSparse.jl) accelerates SparseArrays sparse-dense matrix operations using Intel's MKL library.
+
+5. [SparseArrayKit.jl](https://github.com/Jutho/SparseArrayKit.jl) available for multidimensional sparse arrays.
+
+6. [LuxurySparse.jl](https://github.com/QuantumBFS/LuxurySparse.jl)
```
Comment (Wimmerer) [inline docs/src/index.md:258]:
```suggestion
6. [LuxurySparse.jl](https://github.com/QuantumBFS/LuxurySparse.jl) provides static sparse array formats, as well as a coordinate format.
```

File: docs/src/index.md [hunk 3]
```diff
@@ -249,3 +252,9 @@ Several other Julia packages provide sparse matrix implementations that should b
...
+
+7. [ExtendableSparse.jl](https://github.com/j-fu/ExtendableSparse.jl)
```
Comment (Wimmerer) [inline docs/src/index.md:260]:
```suggestion
7. [ExtendableSparse.jl](https://github.com/j-fu/ExtendableSparse.jl) enables fast insertion into sparse matrices using a lazy approach to new stored indices.
```

**Comment by ViralBShah** [comment]
Merge?

**Pull request merged**